Classes/Controller/LoginController.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+
+namespace Causal\Oidc\Controller;
+
+
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+use TYPO3\CMS\Core\Utility\HttpUtility;
+use TYPO3\CMS\Frontend\ContentObject\ContentObjectRenderer;
+use TYPO3\CMS\Frontend\Controller\TypoScriptFrontendController;
+
+class LoginController
+{
+    /**
+     * Global oidc settings
+     *
+     * @var array
+     */
+    protected $settings;
+
+    /**
+     * TypoScript configuratoin of this plugin
+     *
+     * @var array
+     */
+    protected $pluginConfiguration;
+
+    /**
+     * @var ContentObjectRenderer
+     *
+     * Will be injected automatically, if this controller is called as a plugin
+     */
+    public $cObj;
+
+
+    public function __construct()
+    {
+        $this->settings = unserialize($GLOBALS['TYPO3_CONF_VARS']['EXT']['extConf']['oidc']);
+    }
+
+    /**
+     * @param string $_ ignored
+     * @param array|null $pluginConfiguration
+     *
+     * Main entry point for the OIDC plugin.
+     *
+     * If the user is not logged in, redirect to the authorization server to start the oidc process
+     *
+     * If the user has just been logged in and just came back from the authorization server, redirect the user to the
+     * final redirect URL.
+     */
+    public function login($_ = '', ?array $pluginConfiguration)
```

Comment:
`?array` is not compatible with PHP 5.5 which is still a dependency defined in `composer.json`

Note that I'd be fine to raise the dependency to PHP 7.2, and drop TYPO3 7.6 support as well but this needs to be done separately.